mathics/builtin/plot.py

```diff
@@ -543,6 +634,390 @@ def find_excl(excl):
                           *options_to_rules(options))
 
 
+class BarChart(Builtin):
+    """
+    <dl>
+    <dt>'BarChart[{$p1$, $p2$ ...}]'
+        <dd>draws a bar chart.
+    </dl>
+
+    >> BarChart[{1, 4, 2}]
+     = -Graphics-
+
+    >> BarChart[{1, 4, 2}, ChartStyle -> {Red, Green, Blue}]
+     = -Graphics-
+
+    >> BarChart[{{1, 2, 3}, {2, 3, 4}}]
+     = -Graphics-
+
+    >> BarChart[{{1, 2, 3}, {2, 3, 4}}, ChartLabels -> {"a", "b", "c"}]
+     = -Graphics-
+
+    >> BarChart[{{1, 5}, {3, 4}}, ChartStyle -> {{EdgeForm[Thin], White}, {EdgeForm[Thick], White}}]
+     = -Graphics-
+    """
+
+    from .graphics import Graphics
+
+    attributes = ('HoldAll',)
+
+    options = Graphics.options.copy()
+    options.update({
+        'Axes': '{False, True}',
+        'AspectRatio': '1 / GoldenRatio',
+        'Mesh': 'None',
+        'PlotRange': 'Automatic',
+        'ChartLabels': 'None',
+        'ChartStyle': 'Automatic',
+    })
+
+    requires = (
+        'palettable',
+    )
+
+    def apply(self, points, evaluation, options):
+        '%(name)s[points_, OptionsPattern[%(name)s]]'
+
+        points = points.evaluate(evaluation)
+
+        if points.get_head_name() != 'System`List' or not points.leaves:
+            return
+
+        if points.leaves[0].get_head_name() == 'System`List':
+            if not all(group.get_head_name() == 'System`List' for group in points.leaves):
+                return
+            multiple_colors = True
+            groups = points.leaves
+        else:
+            multiple_colors = False
+            groups = [points]
+
+        def to_number(x):
+            if isinstance(x, Integer):
+                return float(x.get_int_value())
+            y = x.get_float_value()
+            if y is None:
+                return 0.
+            return y
```

Comment:
I think this would be better implemented with `round_to_float(evaluation=evaluation)`. This one doesn't handle non-machine reals and symbolic arguments.